crd/apis/network/v1/network_types.go

```diff
@@ -132,7 +132,13 @@ type NetworkSpec struct {
 	// IPAMMode specifies the IPAM mode for the network.
 	// Valid options include: internal, external
 	// +optional
+	// +deprecatedversion:warning="IPAMMode is deprecated in favor of IPAM4."
 	IPAMMode *IPAMModeType `json:"IPAMMode,omitempty"`
+
+	// IPAM4 specifies the IPAM mode for the network.
+	// Valid options include: Internal, External
+	// +optional
+	IPAM4 *IPAMModeType `json:"ipam4,omitempty"`
```

Comment:
I don't have opinions on how we manage this internally , but just FYI https://github.com/kubernetes/community/blob/master/contributors/devel/sig-architecture/api_changes.md if you want to do this in kubernetes you need to explain how are you going to update the existing clusters, and how are you going to avoid conflicts between both fields and ensure both have the same value